perf/src/sigverify.rs

```diff
@@ -99,20 +103,32 @@ fn verify_packet(packet: &Packet) -> u8 {
         let sig_end = sig_start as usize + size_of::<Signature>();
 
         if pubkey_end >= packet.meta.size || sig_end >= packet.meta.size {
-            return 0;
+            packet.meta.discard = true;
         }
 
         let signature = Signature::new(&packet.data[sig_start..sig_end]);
+
         if !signature.verify(
             &packet.data[pubkey_start..pubkey_end],
             &packet.data[msg_start..msg_end],
         ) {
-            return 0;
+            // Check for tracer pubkey here
+            packet.meta.discard = true;
+            return;
+        }
+
+        if &packet.data[pubkey_start..pubkey_end] == TRACER_KEY.as_ref()
+            // Record the first signature in the transaction
+            && packet.meta.tracer_tx_signature.is_none()
+        {
+            packet.meta.tracer_tx_signature = Some(signature);
         }
+
         pubkey_start += size_of::<Pubkey>();
         sig_start += size_of::<Signature>();
     }
-    1
+
+    packet.meta.discard = false;
```

Comment:
This could potentially cause a discarded packet to be undiscarded. I think it would be better to check above whether the packet is discarded already and only verify if it's not discarded yet just to be safe

Reply:
So all cases where `packet.meta.discard = true` should early return (parallels the `return 0` from the previous code). Also updated to set discard to false at the beginning of the function instead, as a failsafe

Reply:
> Also updated to set discard to false at the beginning of the function instead, as a failsafe

So is it safe to set discard to false at the beginning because we know that `FetchStage` never marks packets as discarded? I think it would be a little safer to just return if discard is true at the top